src/utils.js

```diff
@@ -658,7 +658,9 @@ export default {
         } else if (def.type === 'hasOne') {
           if (localKey) {
             prop.get = function () {
-              return get(this, localKey) ? definition.getResource(relationName).get(get(this, localKey)) : undefined
+              let key = get(this, localKey);
+              let hasKey = isEmpty(key);
```

Comment:
So basically, this line should instead be `let hasKey = (typeof key === 'number' && key >= 0) || (typeof key === 'string' && key)`